### PR TITLE
DEV-5618 Improve ApiFactory initialisation

### PR DESCRIPTION
**Pull request by rizwansaeed**
# Pull Request Checklist

- [x] Read the [contributing guidelines](https://github.com/finbourne/lusid-sdk-python-preview/blob/master/CONTRIBUTING.md)
- [X] Tests pass
- [X] Raised the PR against the `develop` branch

# Description of the PR

Improve ApiFactory initialisation

### CODE REVIEW

File: sdk/lusid/utilities/api_configuration_loader.py [hunk 1]
```diff
@@ -1,55 +1,69 @@
...
+        def vars_from_env():
+            # Load our configuration details from the environment variables
+            return {
+                "token_url": os.getenv("FBN_TOKEN_URL", None),
+                "api_url": os.getenv("FBN_LUSID_API_URL", None),
+                "username": os.getenv("FBN_USERNAME", None),
+                "password": os.getenv("FBN_PASSWORD", None),
+                "client_id": os.getenv("FBN_CLIENT_ID", None),
+                "client_secret": os.getenv("FBN_CLIENT_SECRET", None),
+                "app_name": os.getenv("FBN_APP_NAME", "")
```
Comment (MikeMcGarry) [inline sdk/lusid/utilities/api_configuration_loader.py]: The environment variable names are hard coded both here and in the dictionary lower down. This leaves room for mismatches in the values.

File: sdk/lusid/utilities/api_configuration_loader.py [hunk 2]
```diff
@@ -1,55 +1,69 @@
...
+                var_to_env = {
+                    "token_url": "FBN_TOKEN_URL",
+                    "api_url": "FBN_LUSID_API_URL",
+                    "username": "FBN_USERNAME",
+                    "password": "FBN_PASSWORD",
+                    "client_id": "FBN_CLIENT_ID",
+                    "client_secret": "FBN_CLIENT_SECRET"
+                }
+
+                raise ValueError(f"secrets not specified and missing the following environment variables: {','.join([var_to_env[k] for k in missing.keys()])}")
```
Comment (MikeMcGarry) [inline sdk/lusid/utilities/api_configuration_loader.py]: Can we say "Path to secrets file not specified"

File: sdk/lusid/utilities/api_configuration_loader.py [hunk 3]
```diff
@@ -1,55 +1,69 @@
...
             token_url = os.getenv("FBN_TOKEN_URL", config["api"]["tokenUrl"])
             username = os.getenv("FBN_USERNAME", config["api"]["username"])
-            password = pathname2url(os.getenv("FBN_PASSWORD", config["api"]["password"]))
-            client_id = pathname2url(os.getenv("FBN_CLIENT_ID", config["api"]["clientId"]))
-            client_secret = pathname2url(os.getenv("FBN_CLIENT_SECRET", config["api"]["clientSecret"]))
+            password = os.getenv("FBN_PASSWORD", config["api"]["password"])
+            client_id = os.getenv("FBN_CLIENT_ID", config["api"]["clientId"])
+            client_secret = os.getenv("FBN_CLIENT_SECRET", config["api"]["clientSecret"])
             api_url = os.getenv("FBN_LUSID_API_URL", config["api"]["apiUrl"])
             app_name = os.getenv("FBN_APP_NAME", config["api"].get("applicationName", ""))
```
Comment (MikeMcGarry) [inline sdk/lusid/utilities/api_configuration_loader.py]: I see that we are running os.getenv twice. Here and earlier during our check. Do we want to do this?

File: sdk/lusid/utilities/api_configuration_loader.py [hunk 4]
```diff
@@ -1,55 +1,69 @@
...
 
+            return ApiConfiguration(
+                token_url=token_url,
+                api_url=api_url,
+                username=username,
+                password=password,
+                client_id=client_id,
+                client_secret=client_secret,
+                app_name=app_name
+            )
```
Comment (MikeMcGarry) [inline sdk/lusid/utilities/api_configuration_loader.py]: What happens if there are still missing values at this point? Looks like it won't fail gracefully.

File: sdk/lusid/utilities/api_configuration_loader.py (hunk 1 above)
Comment (rizwansaeed) [reply sdk/lusid/utilities/api_configuration_loader.py]: done

File: sdk/lusid/utilities/api_configuration_loader.py (hunk 2 above)
Comment (rizwansaeed) [reply sdk/lusid/utilities/api_configuration_loader.py]: done

File: sdk/lusid/utilities/api_configuration_loader.py (hunk 3 above)
Comment (rizwansaeed) [reply sdk/lusid/utilities/api_configuration_loader.py]: good spot, copy and paste error 🤦‍♂

File: sdk/lusid/utilities/api_configuration_loader.py (hunk 4 above)
Comment (rizwansaeed) [reply sdk/lusid/utilities/api_configuration_loader.py]: refactored

**Pull request merged**